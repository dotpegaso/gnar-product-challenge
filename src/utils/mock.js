export default {
  cats: [
    {
      id: "dkp",
      photo: 'https://cdn2.thecatapi.com/images/dkp.gif',
      name: 'Fluffy Jenkins',
      age: 3,
      activity: 'Eating',
      peave: 'Water',
      location: 'SPCA Orange County',
      bio: 'Fluffy is a spunky cat who likes to know what is going on at all times. He is curious, smart, and personable. He loves to meow, hunt, eat, and sleep. Catzilla also does not get along with other cats, but he likes dogs. Catzilla is in need of a loving, lasting home.',
      medical: 'https://bit.ly/39BrzIB',
      curious: 10, 
      friendly: 10,
      energetic: 10
    },
    {
      id: "asv",
      photo: 'https://cdn2.thecatapi.com/images/asv.jpg',
      name: 'Floffy Wenkins',
      age: 2,
      activity: 'Run',
      peave: 'Sun',
      location: 'SPCA Orange County',
      bio: '',
      medical: '',
      curious: 10, 
      friendly: 10,
      energetic: 10
    }
  ]
}
