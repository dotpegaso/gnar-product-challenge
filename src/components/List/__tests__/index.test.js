import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import mock from '../../../utils/mock'
import { List } from '..'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: 'dkp',
  }),
  useRouteMatch: () => ({ url: '/details/:id' }),
}))

afterEach(cleanup)

const component = render(
  <Router>
    <List cats={mock.cats} searchString />
  </Router>
)

it('should take a snapshot of List', () => {
  expect(component).toMatchSnapshot()
})

it('should 2 have cats in the list', () => {
  const { getByTestId } = render(
    <Router>
      <List cats={mock.cats} searchString="" />
    </Router>
  )
  expect(getByTestId('cats').children).toHaveLength(2)
})

it('should show the empty list message', () => {
  const { getByTestId } = render(
    <Router>
      <List cats={[]} searchString />
    </Router>
  )
  expect(getByTestId('cats').children).toHaveLength(1)
})
