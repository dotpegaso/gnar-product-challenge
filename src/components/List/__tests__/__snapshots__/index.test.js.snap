// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of List 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="list"
      >
        <h1
          class="title"
        >
          CAT LIST
        </h1>
        <ul
          class="content"
          data-testid="cats"
        >
          <img
            alt="Empty list"
            class="emptyList"
            src="[object Object]"
          />
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="list"
    >
      <h1
        class="title"
      >
        CAT LIST
      </h1>
      <ul
        class="content"
        data-testid="cats"
      >
        <img
          alt="Empty list"
          class="emptyList"
          src="[object Object]"
        />
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
